genesyscloud/routing_queue_outbound_email_address/resource_genesyscloud_routing_queue_outbound_email_address.go

```diff
@@ -0,0 +1,136 @@
+package routing_queue_outbound_email_address
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/mypurecloud/platform-client-sdk-go/v125/platformclientv2"
+	"log"
+	"strings"
+	consistencyChecker "terraform-provider-genesyscloud/genesyscloud/consistency_checker"
+	"terraform-provider-genesyscloud/genesyscloud/provider"
+	resourceExporter "terraform-provider-genesyscloud/genesyscloud/resource_exporter"
+	"terraform-provider-genesyscloud/genesyscloud/util"
+	"terraform-provider-genesyscloud/genesyscloud/util/resourcedata"
+)
+
+/*
+The resource_genesyscloud_routing_queue_outbound-email_address.go contains all the methods that perform the core logic for the resource.
+*/
+
+func getAllAuthRoutingQueueOutboundEmailAddress(ctx context.Context, clientConfig *platformclientv2.Configuration) (resourceExporter.ResourceIDMetaMap, diag.Diagnostics) {
+	resources := make(resourceExporter.ResourceIDMetaMap)
+	proxy := getRoutingQueueOutboundEmailAddressProxy(clientConfig)
+
+	queues, _, err := proxy.getAllRoutingQueues(ctx)
+	if err != nil {
+		return nil, diag.Errorf("failed to get routing queues outbound email address: %s", err)
```

Comment:
Lets use the new util.BuildAPIDiagnostics util so that can get the response and error code in the logs.